tools/run_tests/python_utils/bazel_report_helper.py

```diff
@@ -0,0 +1,252 @@
+                'sleep 60',
+                'PYTHONHTTPSVERIFY=0 python3 ./tools/run_tests/python_utils/upload_rbe_results.py --invocation_id="%s"'
+                % invocation_id,
+                '',
+            ]
+        else:
+            upload_results_lines = []
+
+        outro_lines = [
+            'if [ "$FAILED" != "" ]',
+            'then',
+            '  exit 1',
+            'else',
+            '  # success: plant the pre-generated xml report that says "success"',
+            '  mv -f %s %s' %
+            (success_report_filename, failing_report_filename),
+            'fi',
+        ]
+
+        lines = [
+            line + '\n'
+            for line in intro_lines + upload_results_lines + outro_lines
+        ]
+        f.writelines(lines)
+    os.chmod(bazel_wrapper_filename, 0o775)  # make the unix wrapper executable
+
+    # generate bazel wrapper for windows
+    with open(bazel_wrapper_bat_filename, 'w') as f:
+        intro_lines = [
+            '@echo on',
+            '',
+            'bazel --bazelrc="%s" %%*' % bazel_rc_filename,
+            'set BAZEL_EXITCODE=%errorlevel%',
+            '',
+        ]
+
+        if upload_results:
+            upload_results_lines = [
+                '@rem Sleep to let ResultStore finish writing results before querying',
+                'sleep 60',
+                'python3 tools/run_tests/python_utils/upload_rbe_results.py --invocation_id="%s" || exit /b 1'
+                % invocation_id,
+                '',
+            ]
+        else:
+            upload_results_lines = []
+
+        outro_lines = [
+            'if %BAZEL_EXITCODE% == 0 (',
+            '  @rem success: plant the pre-generated xml report that says "success"',
+            '  mv -f %s %s' %
+            (success_report_filename, failing_report_filename),
+            ')',
+            'exit /b %BAZEL_EXITCODE%',
+        ]
+
+        lines = [
+            line + '\n'
+            for line in intro_lines + upload_results_lines + outro_lines
+        ]
+        f.writelines(lines)
+
+    print('Bazel invocation ID: %s' % invocation_id, file=sys.stderr)
+    print('Upload test results to BigQuery after bazel runs: %s' %
+          upload_results,
+          file=sys.stderr)
+    print('Generated bazel wrapper: %s' % bazel_wrapper_filename,
+          file=sys.stderr)
+    print('Generated bazel wrapper: %s' % bazel_wrapper_bat_filename,
+          file=sys.stderr)
+
+
+# parse command line
+argp = argparse.ArgumentParser(
+    description='Generate bazel wrapper to help with bazel test reports in CI.')
+argp.add_argument(
+    '--report_path',
+    required=True,
+    type=str,
+    help=
+    'Path under which the bazel wrapper and other files are going to be generated'
+)
+argp.add_argument('--report_suite_name',
+                  default='bazel_invocations',
+                  type=str,
+                  help='Test suite name to use in generated XML report')
+args = argp.parse_args()
```

Comment:
optional: I would recommend to move the ArgumentParser portion to the top (for better visibility)

Reply:
done (introducing `__main__` has solved this).